src/app/meetups/meetups.component.ts

```diff
@@ -203,7 +205,17 @@ export class MeetupsComponent implements OnInit, AfterViewInit, OnDestroy {
 
   countSelectedShelf(selected: any) {
     const { inShelf, notInShelf } = this.userService.countInShelf(selected, 'meetupIds');
-    this.selectedJoined = inShelf;
     this.selectedNotJoined = notInShelf;
+    this.selectedJoined = inShelf;
+    let index, meetupDate;
+    for (let i = 0; i < selected.length; i++) {
+      index = this.meetups.data.findIndex(x => x._id === selected[i]);
+      meetupDate = this.meetups.data[index].endDate ? this.meetups.data[index].endDate : this.meetups.data[index].startDate;
+      if (this.dateNow > meetupDate) {
+        this.selectedNotJoined--;
+        selected.splice(i, 1);
+      } 
+    }
+    this.userService.changeShelf(selected, 'meetupIds', 'remove');

```

Comment:
This doesn't actually do anything, and I think you should remove it entirely anyway.  Let's let the user remove themselves from meetups that are older rather than silently removing them in the background.